stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/ThreadRepliesDecorator.kt

```diff
@@ -17,28 +23,187 @@ internal class ThreadRepliesDecorator : BaseDecorator() {
     override fun decoratePlainTextWithFileAttachmentsMessage(
         viewHolder: PlainTextWithFileAttachmentsViewHolder,
         data: MessageListItem.MessageItem
-    ) = Unit
+    ) {
+        setupThreadRepliesView(viewHolder.binding, data)
+    }
 
     override fun decorateOnlyFileAttachmentsMessage(
         viewHolder: OnlyFileAttachmentsViewHolder,
         data: MessageListItem.MessageItem
-    ) = Unit
+    ) {
+        setupThreadRepliesView(viewHolder.binding, data)
+    }
 
     override fun decoratePlainTextWithMediaAttachmentsMessage(
         viewHolder: PlainTextWithMediaAttachmentsViewHolder,
         data: MessageListItem.MessageItem
-    ) = Unit
+    ) {
+        setupThreadRepliesView(viewHolder.binding, data)
+    }
 
     override fun decorateOnlyMediaAttachmentsMessage(
         viewHolder: OnlyMediaAttachmentsViewHolder,
         data: MessageListItem.MessageItem
-    ) = Unit
+    ) {
+        setupThreadRepliesView(viewHolder.binding, data)
+    }
 
     override fun decoratePlainTextMessage(viewHolder: MessagePlainTextViewHolder, data: MessageListItem.MessageItem) {
         setupThreadRepliesView(viewHolder.binding, data)
     }
 
-    override fun decorateGiphyMessage(viewHolder: GiphyViewHolder, data: MessageListItem.MessageItem) = Unit
+    override fun decorateGiphyMessage(viewHolder: GiphyViewHolder, data: MessageListItem.MessageItem) {
+        setupThreadRepliesView(viewHolder.binding, data)
+    }
+
+    private fun setupThreadRepliesView(
+        binding: StreamUiItemMessagePlainTextWithFileAttachmentsBinding,
+        data: MessageListItem.MessageItem
```

Comment:
Looks like we're duplicating a lot of code here in this decorator. Instead of passing in the binding to these methods, you could have a single method and pass in the Views to manipulate (the root, and then the `threadRepliesFootnote`) as two parameters to it.

Take a look at some of the other decorators, like `ReplyDecorator` or `MessageFooterDecorator` for this pattern.

Reply:
I've made the `setupThreadRepliesView()` reusable. Let me know if you can see a better way of doing this.